nmigen/test/test_lib_io.py

```diff
@@ -4,53 +4,59 @@
 from ..back.pysim import *
 from ..lib.io import *
 
+def compare_layout(test_case, layout, expected):
```

Comment:
This could probably be a method of `PinLayoutTestCase`, and the rest could inherit from it.